shaders/src/getters.vs

```diff
@@ -60,18 +61,83 @@ vec3 mulBoneVertex(vec3 v, uint i) {
     return v.x * m[0].xyz + (v.y * m[1].xyz + (v.z * m[2].xyz + m[3].xyz));
 }
 
+void skinPosition(inout vec3 p, const uvec4 ids, const vec4 weights) {
+    // standard skinning for 4 weights, some of them could be zero
+    if (weights.w >= 0.0){
+        p = weights.x * mulBoneVertex(p, uint(ids.x))
+            + weights.y * mulBoneVertex(p, uint(ids.y))
+            + weights.z * mulBoneVertex(p, uint(ids.z))
+            + weights.w * mulBoneVertex(p, uint(ids.w));
+        return;
+    }
+    // skinning for >4 weights
+    vec3 posSum = weights.x * mulBoneVertex(p, uint(ids.x));
+    posSum += weights.y * mulBoneVertex(p, uint(ids.y));
+    posSum += weights.z * mulBoneVertex(p, uint(ids.z));uint pairIndex = -uint(weights.w + 1.);
```

Comment:
Any time you calculate `pairIndex`, you need to make sure the minus is inside the `uint`:

```
uint(-weights.w + 1.)
```

Otherwise `weights.w + 1.` is a negative, and we're casting it to a `uint`.

Reply:
good point, fixed